draft-ietf-taps-interface.md

```diff
@@ -989,6 +990,75 @@ Connection.Batch(
 )
 ~~~
 
+## Send on Active Open: InitiateAndSend {#initiate-and-send}
+
+For application-layer protocols where the Connection initiator also sends the
+first message, the InitiateAndSend() action combines Connection initiation with
+the first Message sent.
+
+Without a message context (as in {{send-basic}}):
+
+~~~
+Connection := Preconnection.InitiateAndSend(messageData)
+~~~
+
+With a message context (as in {{message-props}}):
+
+~~~
+Connection := Preconnection.InitiateAndSend(messageData, messageContext)
+~~~
```

Comment:
I like the general approach of `InitiateAndSend`. It keeps the types right, and avoids issues with trying to call `Send()` on `Preconnection` objects and with enqueued data.

Maybe suggest changing the name to `InitiateAndSendIdempotent()` though, to be explicit? There are security issues with idempotent data, so we need to (try to) make sure people don't use without thinking through the consequences.

Reply:
I  agree, being explicit here would be important.

Reply:
+1 to change to `InitiateAndSendIdempotent()`, since it forces that message property.

Reply:
<bikeshed>How about `InitiateWithIdempotentSend()`?</bikeshed>

Reply:
wfm